engine/execution/state_synchronization/state_diff_storer_test.go

```diff
@@ -1,116 +1,49 @@
 )
 
-const BUCKET_NAME = "flow_public_mainnet14_execution_state"
+func generateExecutionStateDiff(t *testing.T, n int) *ExecutionStateDiff {
+	est := &ExecutionStateDiff{}
+	for i := 0; i < n; i++ {
+		est.TrieUpdates = append(est.TrieUpdates, utils.TrieUpdateFixture(1, 1<<16, 1<<20))
+		collection := unittest.CollectionFixture(1)
+		est.Collections = append(est.Collections, &collection)
+	}
+	return est
+}
 
 func TestStateDiffStorer(t *testing.T) {
-	// this test is intended to be run locally
-	t.Skip("manual test")
+	t.Parallel()
 
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+	bstore, cleanup := test.MakeBlockstore(t, "state-diff-storer-test")
+	defer cleanup()
 
-	client, err := storage.NewClient(ctx, option.WithoutAuthentication())
+	sdp, err := NewStateDiffStorer(&cborcodec.Codec{}, compressor.NewLz4Compressor(), bstore)
 	require.NoError(t, err)
-	bucket := client.Bucket(BUCKET_NAME)
-
-	var maxFileSize int64
-	var maxCollectionsLength int
-	var maxEventsLength int
-	var maxStoreTime time.Duration
-	var maxLoadTime time.Duration
-
-	it := bucket.Objects(ctx, &storage.Query{Prefix: ""})
-	for {
-		attrs, err := it.Next()
-		if err == iterator.Done {
-			break
-		}
-		require.NoError(t, err)
-
-		t.Logf("reading object: %v [%v bytes]\n", attrs.Name, attrs.Size)
-		if attrs.Size > maxFileSize {
-			maxFileSize = attrs.Size
-		}
-
-		reader, err := bucket.Object(attrs.Name).NewReader(ctx)
-		require.NoError(t, err)
-		defer reader.Close()
-
-		bstore, cleanup := test.MakeBlockstore(t, "state-diff-storer-test")
-		defer cleanup()
-
-		sdp, err := NewStateDiffStorer(&cborcodec.Codec{}, compressor.NewLz4Compressor(), bstore)
-		require.NoError(t, err)
-
-		var blockData uploader.BlockData
-
-		decoder := cbor.NewDecoder(reader)
-		require.NoError(t, decoder.Decode(&blockData))
 
-		var collections []*flow.Collection
-		for _, c := range blockData.Collections {
-			collections = append(collections, &flow.Collection{Transactions: c.Transactions})
-		}
-
-		if len(collections) > maxCollectionsLength {
-			maxCollectionsLength = len(collections)
-		}
-		if len(blockData.Events) > maxEventsLength {
-			maxEventsLength = len(blockData.Events)
-		}
-
-		sd := &ExecutionStateDiff{
-			Collections:        collections,
-			TransactionResults: blockData.TxResults,
-			Events:             blockData.Events,
-			TrieUpdate:         blockData.TrieUpdates,
-		}
-
-		start := time.Now()
-		cid, err := sdp.Store(sd)
-		duration := time.Since(start)
-		require.NoError(t, err)
-		t.Logf("time to store state diff: %v\n", duration)
-		if duration > maxStoreTime {
-			maxStoreTime = duration
-		}
-
-		start = time.Now()
-		sd2, err := sdp.Load(cid)
-		duration = time.Since(start)
-		require.NoError(t, err)
-		t.Logf("time to load state diff: %v\n", duration)
-		if duration > maxLoadTime {
-			maxLoadTime = duration
-		}
-
-		assert.True(t, reflect.DeepEqual(sd, sd2))
-	}
+	sd := generateExecutionStateDiff(t, 0)
```

Comment:
wont this generate a diff with no `TrieUpdates`?

Reply:
Yes, it doesn't really matter because from the point of view of the state diff storer, it all just gets serialized to a blob of data in the end anyways.

The main things I want to test here are the case where the state diff is small and all fits into a single block, and the one where it's large and needs to be split into multiple blocks, hence generating intermediate CIDs.